### PR TITLE
Build: don't depend on Interplay

### PR DESCRIPTION
**Pull request by ennru**
Preparations to enable release by tag

* Remove Interplay plugin to disentangle builds
* Copy in parts from Interplay

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -108,16 +99,9 @@ lazy val commonSettings = Def.settings(
...
-    val currentYear = java.time.Year.now(java.time.Clock.systemUTC).getValue
-    Some(
-      HeaderLicense.Custom(
-        s"Copyright (C) 2009-$currentYear Lightbend Inc. <https://www.lightbend.com>"
-      )
-    )
-  },
-  scalaVersion := ScalaVersions.scala212,
-  crossScalaVersions := Seq(ScalaVersions.scala212, ScalaVersions.scala213),
+  headerLicense := Some(HeaderLicense.Custom(s"Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>")),
```
Comment (renatocaval) [inline build.sbt:102]: Good to not depend on java time API.

Btw, in Play and Lagom we followed the Scala project and we removed the years. We only have "Copyright (C) 2009-2020 Lightbend Inc."

File: build.sbt [hunk 2]
```diff
@@ -252,7 +227,8 @@ lazy val `play-jsonJVM` = `play-json`.jvm.settings(
 
 lazy val `play-json-joda` = project
   .in(file("play-json-joda"))
-  .enablePlugins(PlayLibrary)
+  //  .enablePlugins(PlayLibrary)
+  .enablePlugins(Omnidoc, Publish)
```
Comment (renatocaval) [inline build.sbt:231]: PlayLibrary was configuring two settings from Omnidoc, see https://github.com/playframework/interplay/blob/master/src/main/scala/interplay/PlayLibraryBase.scala#L24-L25

We need to add here. I think only the first one is relevant though.

Comment (renatocaval) [reply build.sbt:231]: oh, sorry. I see you have added it later.

File: project/Common.scala [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          s"scm:git:git@github.com:playframework/${repoName}.git"
+        )
+      ),
+      developers += Developer(
+        "contributors",
+        "Contributors",
+        "https://gitter.im/playframework/contributors",
+        url("https://github.com/playframework")
+      ),
+      description := "Play mailer plugin"
```
Comment (renatocaval) [inline project/Common.scala:33]:
```suggestion
      description := "Play JSON"
```

File: build.sbt (hunk 1 above)
Comment (renatocaval) [reply build.sbt:105]: We can move the `commonSettings` to the `Common` plugin. Also let it depend on the Header plugin, like we did for play-soap.

File: build.sbt [hunk 4]
```diff
@@ -139,27 +128,13 @@ lazy val root = project
...
-    )
-  )
+  .settings(publish / skip := true)
 
 lazy val `play-json` = crossProject(JVMPlatform, JSPlatform)
   .crossType(CrossType.Full)
   .in(file("play-json"))
-  .enablePlugins(PlayLibrary, Playdoc)
+  //  .enablePlugins(PlayLibrary, Playdoc)
+  .enablePlugins(Publish, Playdoc)
```
[the remark this answers is not in the document]
Comment (ignasi35) [reply build.sbt:137]: Actually, I think it does need `Omnidoc`. Interplay`s `PlayLibrary` did [bring it in](https://github.com/playframework/interplay/blob/1bcc6a1187d78122fc77f4cc25ee2cf9821053d1/src/main/scala/interplay/PlayLibraryBase.scala#L18).

File: project/Dependencies.scala [hunk 5]
```diff
@@ -0,0 +1,9 @@
+object Dependencies {
+  // scalaVersion needs to be kept in sync with travis-ci
+  val Scala210      = "2.10.7"
```
Comment (ignasi35) [inline project/Dependencies.scala:3]: Is this needed for scalaJS?
I think `interplay` only had `scala210` for `sbt 0.13`...

File: project/Omnidoc.scala [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      val tagged: String      = omnidocTagPrefix.getOrElse("v") + version.value
+      val tree: String        = if (isSnapshot.value) development else tagged
+      val prefix: String      = "/" + (omnidocPathPrefix ?? "").value
+      val path: String = {
+        val buildDir: File      = (baseDirectory in ThisBuild).value
+        val projDir: File       = baseDirectory.value
+        val rel: Option[String] = IO.relativize(buildDir, projDir)
+        rel match {
+          case None if buildDir == projDir => ""                // Same dir (sbt 0.13)
+          case Some("")                    => ""                // Same dir (sbt 1.0)
```
Comment (ignasi35) [inline project/Omnidoc.scala:45]: Probably we can get rid of these two cases in `play-json`

File: build.sbt (hunk 1 above)
Comment (renatocaval) [reply build.sbt:105]: My typo, I meant to remove the year.

Comment (ennru) [reply build.sbt:105]: I'll leave it to keep the delta smaller.

File: build.sbt (hunk 4 above)
[the remark this answers is not in the document]
Comment (ennru) [reply build.sbt]: Added.

File: project/Common.scala (hunk 3 above)
Comment (ennru) [reply project/Common.scala]: 😄

File: project/Dependencies.scala (hunk 5 above)
Comment (ennru) [reply project/Dependencies.scala]: Removed.

File: project/Omnidoc.scala (hunk 6 above)
Comment (ennru) [reply project/Omnidoc.scala:45]: I wanted to keep the diff to Interplay's version minimal.

**Comment by renatocaval** [comment]
Is this ready for merge?

**Pull request merged**